### ISSUE TITLE
New signature: ShouldThrow(Type exceptionType...)

### ISSUE THREAD

**Issue by ewlloyd** [dennisdoomen/FluentAssertions#493]
I propose a new, non-generic signature for ShouldThrow(), to allow (among other things) xUnit users to do something like this:

``` c#
    [Theory]
    [InlineData(typeof(ThisException), "this bogus data")]
    [InlineData(typeof(ThatException), "that bogus data")]
    public void TrulyExceptional(Type exceptionType, string bogusData)
    {
        Action inst = () => new ExceptionHurler(bogusData);
        inst.ShouldThrow(exceptionType);
    }
```

If there are no strenuous objections, I'll work it and put a PR together.

**Comment by somewhatabstract** [comment]
This is a duplicate of https://github.com/dennisdoomen/FluentAssertions/issues/460

**Comment by ewlloyd** [comment]
I'll still see about that PR. It just got easier, though. :-)

**Comment by ewlloyd** [comment]
I'll still see about that PR. It just got easier, though. :-)

On Nov 6, 2016 11:57 PM, "Dennis Doomen" <notifications@github.com> wrote:

Closed #493 <https://github.com/dennisdoomen/FluentAssertions/issues/493>.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/dennisdoomen/FluentAssertions/issues/493#event-849445738>,
or mute the thread
